vignettes/nldi.Rmd

```diff
@@ -0,0 +1,336 @@
+
+If `sf` is not installed locally - or `no_sf = TRUE` - then only the _sourceName_ ad _identifier_ will be returned for LINESTRING objects.
+
+```{r}
+(c101 = findNLDI(comid = 101, no_sf = TRUE))
+```
+
+### "On" network features
+
+Features indexed to the NHDPlus are returned as `POINT` objects. If `sf` is enabled, then the _sourceName_, _identifier_, _X_, _Y_ and _geometry_ (`sfc`) are returned. If `sf` is not locally available, the geometry is dropped but the X and Y values are are retained. 
+
+For example we can return the NLDI representation of USGS NWIS gage [11120000](https://waterdata.usgs.gov/monitoring-location/11120000/#parameterCode=00060) in both an `sf` facilitated and "non-sf" way:
+
+```{r}
+# local sf installation
+findNLDI(nwis = 11120000)
+
+# co
+findNLDI(nwis = 11120000, no_sf = TRUE)
+```
+
+The core feature set of NLDI include the USGS `NWIS` locations, EPA `WQP` locations, and `HUC12` pour points. Each of these available as arguments in `findNLDI` and can be used to request a feature object. Any NLDI feature can be requested by passing a `type`/`ID` pair as a list to the `origin` argument. This will allow the networking capabilities offered in `dataRetrieval` to grow naturally with the NLDI itself.
+
+For example, we can identify both "core", and "non-core" features using the origin argument:
+
+```{r}
+# COMID 101
+findNLDI(origin = list("comid" = '101'))
+
+# Water Data Exchange 2.0 Site CA_45206
+findNLDI(origin = list("wade" = 'CA_45206'))
+```
+
+Again, network features are returned as `LINESTRING`s, indexed features are returned as `POINT`s. If `sf` is enabled, both come with geometries. If `sf` is not enabled, `POINT` objects are returned with their X,Y coordinates and `LINESTRING` (and later `POLYOGON`) geometries only retain the source and ID.
+
+### Location (longitude/latitude)
+
+If you don't know a network or feature ID, a `longitude`/`latitude` (XY) pair can be passed to the `location` argument as a numeric vector (length 2). Doing so will identify the NHDPlus catchment the coordinates fall within and will returned the associated NHDPlusV2 flowline feature.
+
+```{r}
+findNLDI(location = c(-115, 40))
+```
+
+# Navigation
+
+From any feature (`comid`, `huc12`, `nwis`, `wqp`,`origin`) or `location`, four modes of navigation are available including: 
+
+(1) **UT**: Upper Tributary
+(2) **UM**: Upper Mainstream
+(3) **DM**: Downstream Tributary
+(4) **DD**: Downstream Diversions
+
+A example view of these navigation types can be seen below for NWIS site 11109000.
+
+```{r, echo = FALSE, fig.cap= "NLDI Navigation Options"}
+library(ggplot2)
+library(dplyr)
```

Comment:
Need to library() here? Why not do it for sf too or do it for none of them?

Reply:
Here echo is false, needed `ggplot2` for the image and `dplyr` for the pipe (removed that), sf is used with `sf::st_as_sf()`, I didn't need the entire namespace loaded.